### PR TITLE
plutarch-benchmark with perf-diff binary

### PR DESCRIPTION
**Pull request by emiflake**
Draft because I want to clean up some of the diff code + it needs to be wired up to be used in the Hercules effect

Implements #123, #124

### CODE REVIEW

**Comment by L-as** [comment]
@MatthewCroughan ?

**Comment by MatthewCroughan** [comment]
Let's add this binary via a PR, and then let's make a separate PR to add it to the Hercules-CI Effect. But not in the same PR.

**Review by L-as — APPROVED** [review]
LGTM

File: cabal.project [hunk 1]
```diff
@@ -1,3 +1,5 @@
-packages: ./.
+packages:
+  ./.
+  ./plutarch-benchmark/plutarch-benchmark.cabal
```
Comment (srid) [inline cabal.project:3]: I think `./plutarch-benchmark` is equivalent. Even `./*` IIRC.

**Pull request merged**